### DISCUSSION TITLE
Phoniebox and RISC-V SBC

### DISCUSSION THREAD

**Discussion by fritze512** [MiczFlor/RPi-Jukebox-RFID#2186] [General]
Hello,

I'm thinking of getting a RISC-V Single Board Computer, the BeagleV-Fire or the Milk-V Mars.

Would the software work on these computers with no or not much changes?

**Comment by pabera** [comment]
I don't have much experience with those devices. As long as they run Raspberry Pi Debian, I believe it should be fine. GPIO for Phoniebox has a limitation here as it is built for this OS only. DietPi software has been tried, but with mixed results. 

BeagleV-Fire looks like an overkill to me (120+ EUR).
[Milk-V Mars](https://www.golem.de/news/milk-v-mars-der-raspberry-pi-mit-risc-v-prozessor-2306-174689.html) looks like a Pi 3 clone. RPi 3's work really well. Could be worth a try. What would be the benefit over a used Pi 3, for example?

**Reply by fritze512** [reply]
Thanks for the helpful reply.

Yes, BeagleV-Fire is really expensive but Milk-V Mars is in the price range of a RPi. For both there exists a Debian-based GNU-Linux-distro so it could work.

The RPi uses the proprietary instruction set architecture (ISA) ARM while RISC-V is a free and open standard ISA. Since I want to try a RISC-V SBC the phoniebox could be a good first project.

**Reply by pabera** [reply]
I like the idea. It would be great to see how these environments can be supported. We had a discussion in a different forum about support of other setups, incl. DietPi. I would not spend too much on Phoniebox hardware, anything that is more powerful than a Pi3 or Zero 2 is almost a waste of resources unless you do more with it or they get a second life.
